parser/parser.go

```diff
@@ -238,6 +250,7 @@ func (p *parser) stmtsBrace() ast.Stmts {
 
 // Parse a "simple" statement (eg: allowed in a for loop init clause).
 func (p *parser) simpleStmt() ast.Stmt {
+	p.markStartPos()
```

Comment:
So given the above, this would just become `startPos := p.pos`, and the same in `stmt`.